src/requirements/courses-json-generator.ts

```diff
@@ -10,17 +11,26 @@ export type CourseJson = {
 };
 
 const courseJson: CourseJson = Object.fromEntries(
-  Object.entries(filteredAllCourses)
-    .map(([semester, courses]) =>
-      courses.map(
-        course =>
-          [
-            `${course.subject} ${course.catalogNbr}`,
-            { i: course.crseId, t: course.titleLong, r: semester },
-          ] as const
-      )
-    )
-    .flat()
+  Array.from(courseCodeToDataAndOfferedMap.values()).map(
+    ([semester, course]) =>
+      [
+        `${course.subject} ${course.catalogNbr}`,
+        { i: course.crseId, t: course.titleLong, r: semester },
+      ] as const
+  )
 );
 
+const fullCourseJson = (() => {
+  const json: Record<number, Course & { roster: string }[]> = {};
+
+  courseCodeToDataAndOfferedMap.forEach(([roster, course]) => {
+    const existingArray = json[course.crseId] || [];
+    existingArray.push({ ...course, roster });
+    json[course.crseId] = existingArray;
+  });
+
+  return json;
+})();
+
 writeFileSync('src/assets/courses/courses.json', JSON.stringify(courseJson));
```

Comment:
Is courseJson used for our course selector dropdown?

Reply:
Yes, but with full courses it might be needed anymore. I just keep it here to make the PR more focused